src/aws-code-deploy/orb.yml

```diff
@@ -0,0 +1,237 @@
+version: 2.1
+description: "AWS CodeDeploy"
+
+orbs:
+  aws-cli: circleci/aws-cli@dev:snapshot
+
+commands:
+  create-application:
+    parameters:
+      application-name:
+        description:
+          "The name of an AWS CodeDeploy application associated with the applicable IAM user or AWS account."
+        type: string
+    steps:
+      - run:
+          name: ensure-application-created
+          command: |
+            set +e
+            aws deploy get-application --application-name << parameters.application-name >>
+            if [ $? -ne 0 ]; then
+              set -e
+              echo "No application named << parameters.application-name >> found. Trying to create a new one"
+              aws deploy create-application --application-name << parameters.application-name >>
+            else
+              set -e
```

Comment:
Nit: this `set -e` is unnecessary.

Reply:
I guess I just toggled it again a) to be consistent, and b) incase we want to add more to that else branch. Happy to remove it though if you disagree.

Reply:
Consistency makes sense.